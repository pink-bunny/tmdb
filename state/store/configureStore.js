import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
