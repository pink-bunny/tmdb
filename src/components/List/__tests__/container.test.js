import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedList from '../container';

describe('List container', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();
  const props = {
    store,
    removeModalTxt: 'Mock modal text',
    emptyListTxt: 'Mock empty list text',
    fetchList: jest.fn(),
    removeModalAction: jest.fn(),
    loading: true,
    error: '',
    totalItems: 21,
    currentPage: 1
  };
  const wrapper = shallow(<ConnectedList {...props} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */
  const container = wrapper;
  const instance = container.instance();

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('received list', () => {
    wrapper.setProps({
      list: [
        { id: 1, title: 'Mock title' },
        { id: 2, title: 'Mock title' }
      ]
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('componentDidMount() to have been called fetchList()', () => {
    const { fetchList } = wrapper.props();
    instance.componentDidMount();
    expect(fetchList).toHaveBeenCalled();
  });
});
