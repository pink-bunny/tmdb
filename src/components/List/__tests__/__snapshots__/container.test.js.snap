// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List container matches snapshot 1`] = `
<ListComponent
  currentPage={1}
  emptyListTxt="Mock empty list text"
  error=""
  fetchList={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  list={null}
  listTitle="Mock list title"
  loading={true}
  removeModalAction={[MockFunction]}
  removeModalTxt="Mock modal text"
  totalItems={21}
/>
`;

exports[`List container received list 1`] = `
<ListComponent
  currentPage={1}
  emptyListTxt="Mock empty list text"
  error=""
  fetchList={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  list={
    Array [
      Object {
        "id": 1,
        "title": "Mock title",
      },
      Object {
        "id": 2,
        "title": "Mock title",
      },
    ]
  }
  listTitle="Mock list title"
  loading={true}
  removeModalAction={[MockFunction]}
  removeModalTxt="Mock modal text"
  totalItems={21}
/>
`;
